### PR TITLE
Filter interface

### PR DESCRIPTION
**Pull request by alex-r-bigelow**
This branch adds filtering capabilities to the histogram view in the dataset panel.

### CODE REVIEW

**Comment by alex-r-bigelow** [comment]
I accidentally deleted the prior PR for this branch—the gist at this point is that filters are working in the interface, with the following odd behaviors:

- Mongo-backed datasets (example: the `airports` dataset) correctly show the results of filtering in the histograms, however the download endpoint does not function correctly (the downloaded data is wrapped up in an object with additional metadata, and filtering via `extraParameters` is not yet implemented for database assetstores).

- Flat file-backed datasets (example: the `cars` dataset) behave somewhat correctly in that all endpoints work, however our current query language does not support type casting (such that number-based filters end up incorrectly comparing strings).

**Comment by alex-r-bigelow** [comment]
Turns out the original PR wasn't deleted. Reopening #309

**Comment by alex-r-bigelow** [comment]
... and Github isn't letting us reopen #309, so we're moving the discussion here.

**Comment by ronichoudhury** [comment]
Can you remind me what the problem with the checkboxes was?

**Comment by alex-r-bigelow** [comment]
It's a separate issue—we need to implement the ability to specify a `fields` parameter that limits which columns are included in the download (mainly an optimization for really wide datasets).

**Comment by ronichoudhury** [comment]
Ok. I'll QA this in a few minutes.

File: app/resonant-laboratory/stylesheets/pure-css-custom-form-elements/formelements.ai [hunk 1]
```diff
@@ -0,0 +1,439 @@
+%PDF-1.5%����
```
Comment (ronichoudhury) [inline app/resonant-laboratory/stylesheets/pure-css-custom-form-elements/formelements.ai]: Is this a PDF? Does it need to be in source control?

Comment (alex-r-bigelow) [inline app/resonant-laboratory/stylesheets/pure-css-custom-form-elements/formelements.ai]: Shoot, that totally snuck in there. Sorry about that!

Comment (ronichoudhury) [inline app/resonant-laboratory/stylesheets/pure-css-custom-form-elements/formelements.ai]: Can you remove the file from source control?

**Comment by ronichoudhury** [comment]
Once we `git rm` the illustrator file, this LGTM.

**Comment by alex-r-bigelow** [comment]
It's been erased from history 😈 . Merging now.

**Comment by ronichoudhury** [comment]
Thanks @alex-r-bigelow!

:tada: :tada: :tada:

**Pull request merged**